### PR TITLE
Fix linting multiple remotes

### PR DESCRIPTION
**Pull request by mashehu**
Addresses #2052.

### CODE REVIEW

**Comment by awgymer** [comment]
From the error logs:
> LookupError: Could not find branch information for component 
'custom/custom/dumpsoftwareversions'.Please remove the 'modules.json' and rerun 
the command to recreate it

That path looks wrong, like it's derived `custom` as the org path?

**Review by mirpedrol — APPROVED** [review]
LGTM! I also tested it and seems to work nicely 👌

**Comment by awgymer** [comment]
There is another error because `ModulesJson.get_all_components` returns `None` if it would return `{}`. 
There is only one place in the codebase that this is expected:
https://github.com/nf-core/tools/blob/53ccaef64fdb5a9c6ac0b6b8419fa3034c34d5b0/nf_core/components/info.py#L107-L117

But actually if it ever returned `None` the list comprehension would error and `if components is None` would never be reached

**Comment by awgymer** [comment]
Error from `test_modules_lint_patched_modules`:
```
╭─ [!] 4 Module Test Warnings ─────────────────────────────────────────────────╮
│                                           ╷                ╷                 │
│ Module name                               │ File path      │ Test message    │
│╶──────────────────────────────────────────┼────────────────┼────────────────╴│
│ bismark/align                             │ modules/nf-co… │ Conda update:   │
│                                           │                │ bioconda::bism… │
│                                           │                │ 0.23.0 ->       │
│                                           │                │ 0.24.0          │
│ custom/dumpsoftwareversions               │ modules/nf-co… │ New version     │
│                                           │                │ available       │
│ fastqc                                    │ modules/nf-co… │ New version     │
│                                           │                │ available       │
│ custom/dumpsoftwareversions               │ modules/nf-co… │ Process name    │
│                                           │                │ not used for    │
│                                           │                │ versions.yml    │
│                                           ╵                ╵                 │
╰──────────────────────────────────────────────────────────────────────────────╯
╭─ [✗] 1 Module Test Failed ───────────────────────────────────────────────────╮
│                                           ╷                ╷                 │
│ Module name                               │ File path      │ Test message    │
│╶──────────────────────────────────────────┼────────────────┼────────────────╴│
│ bismark/align                             │ modules/nf-co… │ Local copy of   │
│                                           │                │ module does not │
│                                           │                │ match remote    │
│                                           ╵                ╵                 │
╰──────────────────────────────────────────────────────────────────────────────╯
```

File: nf_core/modules/modules_json.py [hunk 1]
```diff
@@ -355,7 +355,10 @@ def find_correct_commit_sha(self, component_type, component_name, component_path
             for commit in modules_repo.get_component_git_log(component_name, component_type, depth=1000)
         )
         for commit_sha in commit_shas:
-            if all(modules_repo.module_files_identical(component_name, component_path, commit_sha).values()):
+            module_dir = modules_repo.get_component_dir(component_name, component_type)
+            if all(
+                modules_repo.module_files_identical(module_dir, component_name, component_path, commit_sha).values()
```
[the remark this answers is not in the document]
Comment (awgymer) [reply nf_core/modules/modules_json.py]: Dropped `component_name` - seems to be working.

**Comment by awgymer** [comment]
`test_modules_lint_no_gitlab` fails because it sees modules:
```
custom/dumpsoftwareversions | /var/folders/q2/785gjdb95c952rl7m18rzmww0000gp/T/tmp7lru83qs/mypipeline/modules/nf-core/custom/dumpsoftwareversions
fastqc | /var/folders/q2/785gjdb95c952rl7m18rzmww0000gp/T/tmp7lru83qs/mypipeline/modules/nf-core/fastqc
multiqc | /var/folders/q2/785gjdb95c952rl7m18rzmww0000gp/T/tmp7lru83qs/mypipeline/modules/nf-core/multiqc
```

It seems that it is fetching all modules regardless of the org?
This makes sense since you are in `self.pipeline_dir`. 
`self.all_remote_modules` is collated as a list of all modules from all remotes.

File: nf_core/modules/nfcore_module.py [hunk 2]
```diff
@@ -38,12 +38,13 @@ def __init__(self, module_name, repo_url, module_dir, repo_type, base_dir, nf_co
         self.is_patched = False
         self.is_patched = None
```
Comment (awgymer) [inline nf_core/modules/nfcore_module.py:39]: I just noticed this duplication of the `is_patched` attribute. Not sure which one should stay?

Comment (mirpedrol) [reply nf_core/modules/nfcore_module.py:39]: Seems like `False` should be the one that makes more sense, as it's supposed to be a bool.

File: nf_core/modules/lint/module_changes.py [hunk 3]
```diff
@@ -39,10 +40,12 @@ def module_changes(module_lint_object, module):
...
         tempdir = module.module_dir
+    module.branch = module_lint_object.modules_json.get_component_branch(
+        "modules", module.module_name, module.repo_url, module.org
+    )
+    modules_repo = nf_core.modules.modules_repo.ModulesRepo(remote_url=module.repo_url, branch=module.branch)
 
-    for f, same in module_lint_object.modules_repo.module_files_identical(
-        module.module_name, tempdir, module.git_sha
-    ).items():
+    for f, same in modules_repo.module_files_identical(module.module_dir, tempdir, module.git_sha).items():
```
Comment (mirpedrol) [inline nf_core/modules/lint/module_changes.py:48]: This `module.module_dir` is a temporary directory, so not the same as we were obtaining before inside this function with `self.get_component_dir(module_name, "modules")`
Going back to passing the module name and obtaining the directory inside the function should make the test pass

Comment (mirpedrol) [reply nf_core/modules/lint/module_changes.py]: For the record, if we read this in the future, `module.module_dir` is the module directory in the pipeline, while what we need is the module directory in the modules repo.

**Review by awgymer — APPROVED** [review]

**Pull request merged**